app/controllers/picture_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PictureController < ApplicationController
+  def show # GET /pictures/:basename
+    compressed_id, extension = params[:basename].split('.')
+    picture = Picture.find_by_id(Picture.uncompress_id(compressed_id))
+    if picture.blank? || extension != picture.extension
+      render :nothing => true, :status => 404
+    else
+      render_picture_content(picture)
+    end
+  end
+
+  private
+
+  def render_picture_content(picture)
+    response.headers['Cache-Control'] = "public"
+    response.headers['Content-Type'] = "image/#{picture.extension}"
+    response.headers['Content-Disposition'] = "inline"
+    render :text => cached_picture_content(picture)
+  end
+
+  def cached_picture_content(picture)
```

Comment:
Since you are pulling back the picture object from the database anyway, what does the cache buy you?  I think we should removing the caching unless you can actually use it.  Perhaps by not querying the content in the initial query?